src/node/blockstorage.cpp

```diff
@@ -371,6 +371,28 @@ bool BlockManager::LoadBlockIndexDB(const Consensus::Params& consensus_params)
     return true;
 }
 
+void BlockManager::ScanAndUnlinkAlreadyPrunedFiles() {
+    AssertLockHeld(::cs_main);
+    if (!m_have_pruned) {
+        return;
+    }
+
+    std::set<int> block_files_to_prune;
+    for (int file_number = 0; file_number < m_last_blockfile; file_number++) {
+        if (m_blockfile_info[file_number].nSize == 0
+            && !CAutoFile(OpenBlockFile(FlatFilePos(file_number, 0), true), SER_DISK, CLIENT_VERSION).IsNull()) {
```

Comment:
This seems likely to be somewhat I/O-heavy.

Can't we just call unlink and ignore any errors?